docs/developer-docs/docs/build-an-app/build-a-custom-smart-contract.md

```diff
@@ -1,13 +1,221 @@
 ﻿---
-sidebar_position: 5
+sidebar_position: 2
 ---
 
 # Build a custom smart contract
 
-*Coming soon*
+## Overview
 
-<!---
+This guide explains how to create and deploy a simple "Hello World" smart contract on the Warden chain. Since it's intended for testing purposes, you'll be running a local chain.
 
-- A step-by-step guide based on a custom example showing unique capabilities of Warden smart contracts. For example, a contract that evaluates Approval Rules.
+## Prerequisites
 
---->
\ No newline at end of file
+Before you start, complete the following prerequisites:
+
+- Install Rust by running the following:
+
+    ```bash
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+    ```
+
+- Set up the CosmWasm development environment:
+
+  - [CosmWasm](https://book.cosmwasm.com/setting-up-env.html): The CosmWasm binary and its dependencies.
+
+  - [cargo-generate](https://cargo-generate.github.io/cargo-generate/installation.html): A tool to help you get up and running quickly with a new Rust project by leveraging a pre-existing git repository as a template.
+
+  - [wasm-opt](https://docs.rs/wasm-opt/latest/wasm_opt/index.html): A tool for optimizing the compiled WebAssembly (Wasm) code.
+   
+   To install these tools, run the following commands:
+
+   ```bash
+   rustup target add wasm32-unknown-unknown
+   cargo install cargo-generate --features vendored-openssl
+   brew install binaryen
+   ```
+
+- [Run a local chain](test/run-a-local-chain) and make sure you have `wardend` correctly installed. You can stop the chain for now if you wish.
+
+   The next steps require your local account name, or key name, which is referenced as `my-key-name` in the provided command-line examples. You can check the list of available keys by executing this command (while the node is running):
```

Comment:
it doesn't require the node to be running

Reply:
> it doesn't require the node to be running

The chain needs to be running only while deploying a contract. We can either do it at the beginning or middle of the tutorial. I think in the start looks clean. But I'm open to moving it to later part of the tutorial too.